classes/webservice/WebserviceRequest.php

```diff
@@ -1622,6 +1622,10 @@ protected function saveEntityFromXml($successReturnCode)
                     }
                     $result = $object->{$objectMethod}();
                     if ($result) {
+                        Hook::exec('actionWebserviceObjectAdd', array(
```

Comment:
There is three problems here for me.
1. the name of the hook is wrong, this can occurred when it's an update
2. what is the interest of having this kind of hook when there are already some in ObjectModel to retrieve the added objects :thinking:  
3. Why calling it `_POST` when it's for `$this->_inputXml`?

Reply:
1. I agree, my fault... name should be without add
2. yes, there are already hooks in the ObjectModel , but I am only interested in changes in webservice with extra parameter $this->_inputXml . For example: there are also hooks such as actionCustomerAccountAdd, actionCustomerAccountUpdate ... instead of using them in the object model actionObjectCustomerAddAfter, actionObjectCustomerUpdateAfter .
3. ok, _inputXml = $this->_inputXml

Reply:
Maybe something like?
1. ```suggestion
                        Hook::exec('actionWebserviceObject' . ucfirst($objectMethod), array(
```
3. `'xml' => $this->_inputXml`

Reply:
1. Yes,  unfortunately, objectMethod varies depending on whether there is a special function for Webservice, add or update for any object that can be changed to addWs, updateWs ....

protected $webserviceParameters = array(
         'objectMethods' => array(
             'add' => 'addWs'
             'update' => 'updateWs'
         )
So I suggest to restore objectMethod before calling the hook

// Call alternative method for add / update
$objectMethod = ($this-> method == 'POST' ? 'add' : 'update');

we can leave the name general or add the name of the object 
'actionWebserviceObject' .  get_class($object) . ucfirst($objectMethod)